cumulusci/tasks/metadata_etl/base.py

```diff
@@ -258,9 +258,26 @@ def _transform(self):
 
         extension = configuration["extension"]
         directory = entity_configurations[0]
+        source_metadata_dir = self.retrieve_dir / directory
+
+        if "*" in self.api_names:
+            # Walk the retrieved directory to get the actual suite
+            # of API names retrieved and rebuild our api_names list.
+            self.api_names.remove("*")
+            self.api_names = self.api_names.union(
+                unquote(metadata_file.stem)
+                for metadata_file in source_metadata_dir.iterdir()
+                if metadata_file.suffix == f".{extension}"
+            )
+
+        removed_api_names = set()
 
         for api_name in self.api_names:
-            path = self.retrieve_dir / directory / f"{api_name}.{extension}"
+            # Page Layout names can contain spaces, but parentheses are quoted.
+            # Those files _cannot_ be deployed back into Salesforce under their
+            # quoted names - we have to rename them.
+            quoted_api_name = quote(api_name, safe=" ")
```

Comment:
I _think_ this is the right way to handle metadata names. For most components, the API name is required to be [A-Za-z0-9_]; Page Layouts can include parens and other characters which are URL-encoded, but spaces are not.

(@jstvz have you seen any other variants to be aware of?)